internal/core/api.rs

```diff
@@ -591,7 +591,7 @@ mod weak_handle {
     /// as the one it has been created from.
     /// This is useful to use with [`invoke_from_event_loop()`] or [`Self::upgrade_in_event_loop()`].
     pub struct Weak<T: ComponentHandle> {
-        inner: vtable::VWeak<ComponentVTable, T::Inner>,
+        pub(crate) inner: vtable::VWeak<ComponentVTable, T::Inner>,
```

Comment:
Please make a pub(crate) accessor instead.

The reason why this is in a module is to encapsulate the unsafe code inside the module

But by making this pub(crate), this becomes unsound as code from the crate can change the vweak with something that belong in another thread.